cli/dcoscli/task/main.py

```diff
@@ -148,6 +151,18 @@ def _log(follow, completed, lines, task, file_):
                 raise DCOSException(msg)
         raise DCOSException('No matching tasks. Exiting.')
 
+    if file_ in ('stdout', 'stderr') and log.dcos_log_enabled():
+        try:
+            _dcos_log(follow, tasks, lines, file_, completed)
+            return 0
+        except DCOSException as e:
```

Comment:
`DCOSAuthenticationException` and `DCOSAuthorizationException` should be caught first because they are also of type `DCOSException`